examples/nlp/language_modeling/megatron_change_num_partitions.py

```diff
@@ -1384,6 +1385,38 @@ def main():
                 with open_dict(model.cfg):
                     model.cfg.tokenizer.model = args.tokenizer_model_path
 
+            else:
+                # Extract all artifacts that are not binary files
+                # and save them to the model directory
+                logging.info("Extracting artifact from NeMo file...")
```

Comment:
Could this logic be moved outside of the `for pp_rank` loop so that it's run only once?

(also, ideally what comes after the extraction code would be in a `try...finally` block so that `temp_dir` is always deleted on exit)

Reply:
Not really. This logic is shared for all TP PP and must execute in a loop.
It should be fast. The extraction only works on the base level of the nemo file and does not recurse the actual checkpoint dirs.

Reply:
No. Simply calling tarfile open does not untar the file - it only opens a object that can traverse the tar object, not it's contents inside it. You need to explicitly call member.extractfile() to get the file pointer to the actual object

Reply:
I quoted that line to show that it was the same file, the actual extraction occurs below in the code. My point is that the same file(s) are extracted for each PP rank which seems pointless, it would be cleaner to do this only once before the `for` loop.